### PR TITLE
CodeGen: Remove -disable-debug-info-print cl::opt

### PR DESCRIPTION
**Pull request by arsenm**
This was first introduced way back in in 2010 by
6c74a872a8d34d41b751efb68e335cbe91b5a5cc, and has little evidence
of use. Only one test attempts to make use of this, but it's
also redundant since it's also using strip to drop debug info anyway
(and that also makes the test buggy, since it's intended to test
with and without debug info).

The other tests using it were only added to test the option after
discovering it was untested and moved, in later commits.

### CODE REVIEW

**Comment by arsenm** [comment]
* **#100319** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100319?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by dwblaikie** [comment]
Generally seems OK to me - though did find this https://github.com/llvm/llvm-project/issues/44225 recent (well, 2020) issue using it to debug debug-info-affects-codegen bugs, which makes sense (can straight up diff the output, since both sides don't have any debug info, which is a bit easier to check than other ways).

Myabe @jmorse and co might have opinions on whether they want this sort of thing for any ongoing work on the debug info intrinsics migration, etc.

**Comment by vtjnash** [comment]
I have never used it or needed it, so I am not a good reviewer for it. I had thought it was a strange feature flag, but I had been forced to preserve it when I touched code in that area because of both tests and user comments when I accidentally broke it in https://reviews.llvm.org/D89613#2336529. So I am in favor of removing it, as it seems undocumented and minimally tested, but I was aware that some folks seemed to find it useful.

**Review by s-barannikov — APPROVED** [review]
Makes sense to me

File: llvm/lib/CodeGen/MachineModuleInfo.cpp [hunk 1]
```diff
@@ -31,10 +31,6 @@
 using namespace llvm;
 using namespace llvm::dwarf;
 
-static cl::opt<bool>
```
Comment (s-barannikov) [inline llvm/lib/CodeGen/MachineModuleInfo.cpp:34]: CommandLine.h include appears to be no longer needed.

**Pull request merged**